File: 백준/Silver/31246. 모바일 광고 입찰/모바일 광고 입찰.py
import sys
input = sys.stdin.readline
N,K = map(int, input().split())

win0 = 0
pos = []

for i in range(N):
    A,B = map(int, input().split())
    d = B - A
    if d<= 0:
        win0 +=1
    else:
        pos.append(d)
if win0 >= K:
    print(0)
else:
    need = K- win0
    pos.sort()
    print(pos[need-1])

File: 백준/Silver/31246. 모바일 광고 입찰/모바일 광고 입찰.js
const fs = require("fs");
const data = fs.readFileSync(0, "utf8").trim().split(/\s+/).map(Number);

let idx =0
const N = data[idx++], K = data[idx++]

let win0 =0
let pos = []

for(let i=0;i<N;i++){
    const A = data[idx++], B=data[idx++]
    const d= B-A
    if (d<=0) win0++
    else pos.push(d)
}

if(win0>=K){
    console.log(0)
} else {
    const need = K -win0
    pos.sort((a,b)=> a-b)
    console.log(pos[need-1])
    
}